db/change_cache_test.go

```diff
@@ -226,6 +226,126 @@ func TestLateSequenceHandlingWithMultipleListeners(t *testing.T) {
 
 }
 
+// Verify that a continuous changes feed hitting an error when building its late sequence feed will roll back to
+// its low sequence value, then recover and successfully send subsequent late sequences.
+func TestLateSequenceErrorRecovery(t *testing.T) {
+
+	if base.TestUseXattrs() {
+		t.Skip("This test does not work with XATTRs due to calling WriteDirect().  Skipping.")
+	}
+
+	defer base.SetUpTestLogging(base.LevelInfo, base.KeyChanges|base.KeyCache)()
+
+	db, testBucket := setupTestDBWithCacheOptions(t, shortWaitCache())
+	defer tearDownTestDB(t, db)
+	defer testBucket.Close()
+
+	db.ChannelMapper = channels.NewDefaultChannelMapper()
+
+	// Create a user with access to channel ABC
+	authenticator := db.Authenticator()
+	assert.True(t, authenticator != nil, "db.Authenticator() returned nil")
+	user, err := authenticator.NewUser("naomi", "letmein", channels.SetOf(t, "ABC"))
+	assert.NoError(t, err, fmt.Sprintf("Error creating new user: %v", err))
```

Comment:
`assert.NoError` will print the error if it got one - we should probably treat as a fatal error too

```suggestion
	require.NoError(t, err, "Unexpected error creating user")
```